falcon/routing/static.py

```diff
@@ -49,6 +55,10 @@ def __init__(self, prefix, directory, downloadable=False):
         self._prefix = prefix
         self._directory = directory
         self._downloadable = downloadable
+        if default_filename is not None:
+            self._default_filename = os.path.join(directory, default_filename)
```

Comment:
Should it be possible for subdirectories to provide their own versions of this file? If so, the path will have to be joined on the fly, rather than up front. 

For example, suppose the prefix is `/foo/bar` and a request comes in for `/foo/bar/baz/`. You might want that to map to `/foo/bar/baz/index.html`, while a request for `/foo/bar/` would map to `/foo/bar/index.html`.

Reply:
The use case I wanted to cover was SPA, where both `/foo/bar` and `/foo/bar/baz/` would need to map to `/foo/bar/index.html` (supposing the SPA home is `/foo/bar`)